skrub/_fuzzy_join.py

```diff
@@ -183,9 +183,10 @@ def _nearest_matches(
     neigh.fit(aux_array)
     distance, neighbors = neigh.kneighbors(main_array, return_distance=True)
     idx_closest = np.ravel(neighbors)
-    distance = distance / np.max(distance)
-    # Normalizing distance between 0 and 1:
-    matching_score = 1 - (distance / 2)
+    max_dist = distance.max()
+    if max_dist != 0:
```

Comment:
Is it safe to compare floats without any tolerance?
Wouldn’t something like this be more preferable https://numpy.org/doc/stable/reference/generated/numpy.isclose.html

Reply:
I think an exact comparison is ok here because it is just to avoid a division by 0